es-vue-base/src/lib-components/EsFileUpload.vue

```diff
@@ -0,0 +1,230 @@
+                    Browse Files
+                </es-button>
+                <es-button
+                    class="d-inline-block d-md-none w-100 mb-2"
+                    variant="secondary"
+                    outline
+                    @click="openFilePicker">
+                    Choose File
+                </es-button>
+            </div>
+        </div>
+        <div>
+            <slot name="helpText" />
+        </div>
+        <b-form-file
+            ref="fileInput"
+            v-model="pickedItems"
+            :state="Boolean(pickedItems)"
+            :accept="fileTypesAsString"
+            style="display: none"
+            multiple />
+    </div>
+</template>
+
+<script lang="js">
+import { BFormFile } from 'bootstrap-vue';
+import EsButton from './EsButton.vue';
+import IconUpload from '../lib-icons/icon-upload.vue';
+
+export default {
+    name: 'EsFileUpload',
+    components: {
+        BFormFile,
+        EsButton,
+        IconUpload,
+    },
+    props: {
+        uploadUrls: {
+            type: Array,
+            default: () => [],
+        },
+        fileTypes: {
+            type: Array,
+            default: () => [],
+        },
+        maxFileSize: {
+            type: Number,
+            default: 100,
+            required: false,
+        },
+        uploadSuccessStatusCode: {
+            type: Number,
+            default: 200,
+            required: false,
+        },
+        collapsed: {
+            type: Boolean,
+            default: false,
+            required: false,
+        },
+    },
+    data() {
+        return {
+            pickedItems: [],
+            files: [],
+            fileUrls: [], // For thumbnail preview
+            active: false,
+        };
+    },
+    computed: {
+        fileTypesAsString() {
+            return this.fileTypes.join(', ');
+        },
+    },
+    watch: {
+        uploadUrls: {
+            handler(newUrls) {
+                if (newUrls.length > 0 && newUrls.length === this.files.length) this.upload();
+            },
+            deep: true,
+        },
+        files: {
+            handler(newFiles) {
+                if (newFiles.length > 0 && newFiles.length === this.uploadUrls.length) this.upload();
+            },
+            deep: true,
+        },
+        pickedItems(newVal) {
+            this.files = newVal;
+            if (newVal.length > 0 && this.fileSizeValid()) {
+                this.$emit('readyToUpload', this.files.length);
+                this.readFilesIntoUrl(newVal);
+            }
+        },
+    },
+    methods: {
+        fileSizeValid() {
+            // Check if the combined file sizes exceed the maxFileSize prop
+            const totalFileSize = this.files.reduce((acc, file) => acc + file.size, 0);
+            if (totalFileSize > this.maxFileSize * 1000000) {
+                this.$emit('fileSizeError');
+                return false;
+            }
+            return true;
+        },
```

Comment:
Why would the combined size matter? I would think we should be enforcing limits on individual files.

Reply:
Agreed